gemrb/GUIScripts/GUIMA.py

```diff
@@ -24,282 +24,140 @@
-	OptionsWindow.SetFrame ()
+	Window = OpenMapWindow()
 
 	# World Map
 	Button = Window.GetControl (1)
 	Button.SetState (IE_GUI_BUTTON_LOCKED)
 
+	Map = Window.GetControl (2)
+
 	# Hide or Show mapnotes
 	if HasMapNotes ():
 		Button = Window.GetControl (3)
 		Button.SetState (IE_GUI_BUTTON_LOCKED)
 
 		Label = Window.GetControl (0x10000003)
 		Label.SetText ("")
-
-	# Map Control
-	if GameCheck.IsBG2() or GameCheck.IsIWD2():
-		Window.CreateMapControl (2, 0, 0, 0, 0, 0x10000003, "FLAG1")
-	else:
-		Window.CreateMapControl (2, 0, 0, 0, 0)
-	Map = Window.GetControl (2)
-	if HasMapNotes ():
-		GemRB.SetVar ("ShowMapNotes",IE_GUI_MAP_REVEAL_MAP)
+	
 		Map.SetVarAssoc ("ShowMapNotes", IE_GUI_MAP_REVEAL_MAP)
+
 	Map.SetEvent (IE_GUI_MAP_ON_PRESS, RevealMap)
-	Window.SetVisible (WINDOW_VISIBLE)
-	OptionsWindow.SetVisible (WINDOW_GRAYED)
-	PortraitWindow.SetVisible (WINDOW_GRAYED)
-	OptionsWindow.SetVisible (WINDOW_FRONT)
-	PortraitWindow.SetVisible (WINDOW_FRONT)
-	Window.SetVisible (WINDOW_FRONT)
-	if HasMapNotes ():
-		Map.SetStatus (IE_GUI_CONTROL_FOCUSED|IE_GUI_MAP_REVEAL_MAP)
-	else:
-		Map.SetStatus (IE_GUI_CONTROL_FOCUSED)
+
+	Window.ShowModal (MODAL_SHADOW_GRAY)
 	GemRB.GamePause (0,0)
 	return
 
 ###################################################
-def OpenMapWindow ():
-	import GameCheck
-	global MapWindow, OptionsWindow, PortraitWindow
-	global OldPortraitWindow, OldOptionsWindow
-
-	if GUICommon.CloseOtherWindow (OpenMapWindow):
-		if MapWindow:
-			MapWindow.Unload ()
-		if OptionsWindow:
-			OptionsWindow.Unload ()
-		if PortraitWindow:
-			PortraitWindow.Unload ()
-
-		MapWindow = None
-		#this window type should block the game
-		GemRB.SetVar ("OtherWindow", -1)
-		GUICommon.GameWindow.SetVisible(WINDOW_VISIBLE)
-		GemRB.UnhideGUI ()
-		GUICommonWindows.PortraitWindow = OldPortraitWindow
-		OldPortraitWindow = None
-		GUICommonWindows.OptionsWindow = OldOptionsWindow
-		OldOptionsWindow = None
-		GUICommonWindows.SetSelectionChangeHandler (None)
-		return
-
-	GemRB.HideGUI ()
-	GUICommon.GameWindow.SetVisible(WINDOW_INVISIBLE)
-
-	GemRB.LoadWindowPack ("GUIMAP", WIDTH, HEIGHT)
-	MapWindow = Window = GemRB.LoadWindow (2)
-	#this window type blocks the game normally, but map window doesn't
-	GemRB.SetVar ("OtherWindow", MapWindow.ID)
-	#saving the original portrait window
-	OldPortraitWindow = GUICommonWindows.PortraitWindow
-	PortraitWindow = GUICommonWindows.OpenPortraitWindow ()
-	OldOptionsWindow = GUICommonWindows.OptionsWindow
-	OptionsWindow = GemRB.LoadWindow (0)
-	GUICommonWindows.MarkMenuButton (OptionsWindow)
-	GUICommonWindows.SetupMenuWindowControls (OptionsWindow, GameCheck.IsIWD2(), OpenMapWindow)
-	OptionsWindow.SetFrame ()
+
+def InitMapWindow (Window):
+	global MapWindow
+	MapWindow = Window
 
 	# World Map
 	Button = Window.GetControl (1)
-	Button.SetEvent (IE_GUI_BUTTON_ON_PRESS, OpenWorldMapWindowInside)
+	Button.SetEvent (IE_GUI_BUTTON_ON_PRESS, lambda: OpenWorldMapWindow())
```

Comment:
Codacy found an issue: [Lambda may not be necessary](https://app.codacy.com/manual/gemrb/gemrb/pullRequest?prid=6310305)